From 8d8699ade7c3bd0118934f1f234b8c7741219828 Mon Sep 17 00:00:00 2001
From: Chen-Xintong <xintong.chen@intel.com>
Date: Fri, 25 Feb 2022 17:40:13 +0800
Subject: Using errors library API instead of DeepEqual as the judgment
 condition (#37519)

* use errors.Is instead of DeepEqual

* update

* update

* use errEqual instead error.Is
---
 security/pkg/k8s/tokenreview/k8sauthn_test.go | 12 +++++++++++-
 security/pkg/pki/util/generate_csr_test.go    |  2 +-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/security/pkg/k8s/tokenreview/k8sauthn_test.go b/security/pkg/k8s/tokenreview/k8sauthn_test.go
index b3d3f6b7b9..ea18f91c20 100644
--- a/security/pkg/k8s/tokenreview/k8sauthn_test.go
+++ b/security/pkg/k8s/tokenreview/k8sauthn_test.go
@@ -108,8 +108,18 @@ func TestGetTokenReviewResult(t *testing.T) {
 		if !reflect.DeepEqual(result, tc.expectedResult) {
 			t.Errorf("TestGetTokenReviewResult failed: case: %q, actual result is %v, expected is %v", tc.name, result, tc.expectedResult)
 		}
-		if !reflect.DeepEqual(err, tc.expectedError) {
+		if !errEqual(err, tc.expectedError) {
 			t.Errorf("TestGetTokenReviewResult failed: case: %q, actual error is %v, expected is %v", tc.name, err, tc.expectedError)
 		}
 	}
 }
+
+func errEqual(err error, target error) bool {
+	if target == nil {
+		return err == target
+	} else if err == nil {
+		return false
+	} else {
+		return err.Error() == target.Error()
+	}
+}
diff --git a/security/pkg/pki/util/generate_csr_test.go b/security/pkg/pki/util/generate_csr_test.go
index 58db3a8a8a..211af132c6 100644
--- a/security/pkg/pki/util/generate_csr_test.go
+++ b/security/pkg/pki/util/generate_csr_test.go
@@ -59,7 +59,7 @@ func TestGenCSR(t *testing.T) {
 		csrPem, _, err := GenCSR(tc.csrOptions)
 		if err != nil {
 			if tc.err != nil {
-				if reflect.DeepEqual(err, tc.err) {
+				if err.Error() == tc.err.Error() {
 					continue
 				}
 				t.Fatalf("%s: expected error to match expected error: %v", id, err)
-- 
2.35.3

